src/Domain/TagsTimeDependent.cpp

```diff
@@ -0,0 +1,114 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/TagsTimeDependent.hpp"
+
+#include <memory>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/Domain.hpp"
+#include "NumericalAlgorithms/LinearOperators/Divergence.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace domain {
+namespace Tags {
+template <size_t Dim>
+void InertialFromGridCoordinatesCompute<Dim>::function(
+    const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
+        target_coords,
+    const tnsr::I<DataVector, Dim, Frame::Grid>& source_coords,
+    const boost::optional<std::tuple<
+        tnsr::I<DataVector, Dim, Frame::Inertial>,
+        ::InverseJacobian<DataVector, Dim, Frame::Grid, Frame::Inertial>,
+        ::Jacobian<DataVector, Dim, Frame::Grid, Frame::Inertial>,
+        tnsr::I<DataVector, Dim, Frame::Inertial>>>&
+        grid_to_inertial_quantities) noexcept {
+  if (not grid_to_inertial_quantities) {
+    for (size_t i = 0; i < Dim; ++i) {
+      target_coords->get(i).set_data_ref(
+          // NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
+          &const_cast<DataVector&>(source_coords.get(i)));
```

Comment:
please explain (probably in code comment as well) why this isn't hazardous. Same goes for below scary const casts